### ISSUE TITLE
Product Import does not update product if sku has white space

### ISSUE THREAD

**Issue by DeanForde** [woocommerce/woocommerce#29435]
in a rush, so excuse the ticket, however, I had 6000+ products in a csv that I downloaded from a wholesale site, I wanted to modify the prices for a retail site so I made the changes in the csv and uploaded to my retail site only to find some of the products updated and some didn't. After further inspection I noticed that it seems to be that some of the products have custom SKU's where the format of the SKU might be as follows "123456 Oaktree". I then noticed it was all the SKU's that contain white space that did not update through the standard Woocommerce core product importer. Would I be correct that this is the cause? I am also not to sure if this is an issue or expected behavior, expected I presume as generally an SKU wouldn't have white space? but could be frustrating for users who are unaware in the future.

**Comment by juliaamosova** [comment]
Hi @DeanForde,

Thanks for raising the issue!

>I then noticed it was all the SKU's that contain white space that did not update through the standard Woocommerce core product importer. Would I be correct that this is the cause? I am also not to sure if this is an issue or expected behavior, expected I presume as generally an SKU wouldn't have white space?

I believe space is not allowed in SKU, however, I am adding `needs developer feedback` label on this issue to get a confirmation from one of the Core developers.

Once we get the confirmation, we should update the docs to include the right answer: 

https://docs.woocommerce.com/document/managing-products/#section-5

**Comment by roykho** [comment]
Hi @DeanForde - So I've tested this and was not able to replicate the issues with SKU having spaces. Generally speaking it is best to have SKUs that don't have a space, instead use hyphens. However WC does allow spaces in SKUs. Make sure you enable this setting where it will either match the ID or SKU. ![](https://d.pr/i/Do8aUN+)

Perhaps the issue you're seeing is caused by something else?

**Comment by DeanForde** [comment]
Hi Julia,

Yes that is correct. However I do agree, I don't think an SKU should
contain whitespace, but I am currently working with a business owner where
this is the case.
From their understanding, there is no rule to say that they can't or
shouldn't have whitespace in SKU's, I mean I wasn't aware that this would
be an issue and it was only when I started using the API that I noticed the
issue (after some time of trying to figure out what was actually causing
the issue). Furthermore does this mean that Woocommerce was not updating
the products on the Wordpress/Woocommerce backend/UI either as the SKU's
had whitespaces in them? or is this behavior specific to just the API?  I
don't think it should be an issue to allow for this behavior though
as essentially it's just a string? But again I would be in agreement that
as a rule of thumb SKU's should not consist of white spaces. If this
behavior is not allowed then maybe some field validation on the UI at
least? I am not sure how you would manage it as regards csv uploads, but
you see where the confusion is hopefully

Kindest regards,
Dean Forde
